### PR TITLE
add rabbitmq to semantic conventions yaml

### PR DESCRIPTION
**Pull request by blumamir**
In trace semantic conventions markdown there is a [section about rabbitmq](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#rabbitmq) with attribute `messaging.rabbitmq.routing_key`.
This attribute is not reflected in the `messaging.yaml` file [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/messaging.yaml).

opentelemetry-js auto-generate the attributes from this yaml, thus it is [currently missing](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-semantic-conventions/src/trace/SemanticAttributes.ts)

### CODE REVIEW

**Comment by blumamir** [comment]
I think the `docfx` check failure is not related to this PR: 
```
docfx not installed. An error occurred during installation:
 The remote server returned an error: (503) Server Unavailable. Service Unavailable
docfx package files install completed. Performing other installation steps.
```

**Comment by arminru** [comment]
Good catch, thank you!

Please make sure to reference this new group in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#rabbitmq and run the markdown generator as documented here: https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions.
This way, we can also remove the sentence "_The routing key MUST be provided to the attribute `messaging.rabbitmq.routing_key`, unless it is empty._" from the prose.

File: semantic_conventions/trace/messaging.yaml [hunk 1]
```diff
@@ -121,6 +121,19 @@ groups:
...
+      prefix: messaging.rabbitmq
+      extends: messaging
+      brief: >
+        Attributes for RabbitMQ
+      attributes:
+        - id: routing_key
+          type: string
+          brief: >
+            In RabbitMQ, the destination is defined by an exchange and a routing key.
+            The routing key MUST be provided to this attribute, unless it is empty.
```
Comment (arminru) [inline semantic_conventions/trace/messaging.yaml:134]: There is a built-in syntax or this:

```suggestion
          required:
            conditional: Unless it is empty.
```

**Comment by arminru** [comment]
> I think the `docfx` check failure is not related to this PR:
> 
> ```
> docfx not installed. An error occurred during installation:
>  The remote server returned an error: (503) Server Unavailable. Service Unavailable
> docfx package files install completed. Performing other installation steps.
> ```

Indeed. I re-ran the job and now it's passing. Apparently Chocolatey was down before.

File: semantic_conventions/trace/messaging.yaml (hunk 1 above)
Comment (blumamir) [reply semantic_conventions/trace/messaging.yaml]: done

**Comment by blumamir** [comment]
> Good catch, thank you!
> 
> Please make sure to reference this new group in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#rabbitmq and run the markdown generator as documented here: https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions.
> This way, we can also remove the sentence "_The routing key MUST be provided to the attribute `messaging.rabbitmq.routing_key`, unless it is empty._" from the prose.

Done. Thank you for the review and for pointing me to the relevant documentation.
Great automation 👍

File: semantic_conventions/trace/messaging.yaml [hunk 2]
```diff
@@ -121,6 +121,20 @@ groups:
...
+    - id: messaging.rabbitmq
+      prefix: messaging.rabbitmq
+      extends: messaging
+      brief: >
+        Attributes for RabbitMQ
+      attributes:
+        - id: routing_key
+          type: string
+          required:
+            conditional: Unless it is empty.
```
Comment (arminru) [inline semantic_conventions/trace/messaging.yaml:133]:
```suggestion
            conditional: Unless it is empty.
```

**Review by arminru — APPROVED** [review]
Please also add an entry in CHANGELOG.md that this already specified key is now also defined in the respective YAML file.

**Comment by blumamir** [comment]
> Please also add an entry in CHANGELOG.md that this already specified key is now also defined in the respective YAML file.

done

**Review by carlosalberto — APPROVED** [review]

**Review by SergeyKanzhelev — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
this PR is uncontroversial as it only changes from text to yaml, not the semantic. And got a lot of approvals fast. Merging

**Pull request merged**